hack/lib/version.sh

```diff
@@ -36,6 +36,16 @@ kube::version::get_version_vars() {
     return
   fi
 
+  # If the kubernetes source was exported through git archive, then
+  # we likely don't have a git tree, but these magic values may be filled in.
+  if [[ '$Format:%%$' == "%" ]]; then
+    KUBE_GIT_COMMIT='$Format:%H$'
+    KUBE_GIT_TREE_STATE="not a git tree"
+    if [[ '$Format:%D$' =~ tag:\ (v[^ ]+) ]]; then
```

Comment:
we may want to leave a comment on why this works, something like
"If the archive was created at a git tag `vX.Y.Z`, then the ref names will include `tag: vX.Y.Z`"

Reply:
Good suggestion.  Done.